### PR TITLE
ci(angular): fail build when an angular directive or component is not exported

### LINKED ISSUE #548: Make sure the build fails if an Angular directive or component is not exported in the main module
**Description**
We tend to forget to export the newly created directives / components in the main angular module, which should be mandatory.
In order to avoid this, we should either make this angular module generated automatically or add a script that will fail the ci if it is missing.

### PR DESCRIPTION
**Pull request by quentinderoubaix**
closes #548

### CODE REVIEW

File: angular/lib/scripts/verify-module.mjs [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import {glob} from 'glob';
+import path from 'path';
+import {fileURLToPath} from 'url';
+import {readFile, writeFile, mkdir, rm} from 'fs/promises';
```
Comment (MarkoOleksiyenko) [inline angular/lib/scripts/verify-module.mjs:4]: writeFile, mkdir, rm are not used

File: angular/lib/scripts/verify-module.mjs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+
+const pathRegExp = /\\/g;
+export function normalizePath(str) {
```
Comment (MarkoOleksiyenko) [inline angular/lib/scripts/verify-module.mjs:9]: do we need to export the function here?

File: angular/lib/scripts/verify-module.mjs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	return str.replace(pathRegExp, '/');
+}
+
+const files = (
+	await glob('**/*.component.ts', {
+		cwd: path.join(__dirname, '../src/components'),
```
Comment (MarkoOleksiyenko) [inline angular/lib/scripts/verify-module.mjs:15]: minor: i feel every time that the string is repeated 2+ times it is better to create a var for it (../src/components)

File: angular/lib/scripts/verify-module.mjs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+const COMPONENTS_FOLDER = path.join(__dirname, '../src/components');
+
+const files = (
+	await glob('**/*.component.ts', {
+		cwd: COMPONENTS_FOLDER,
+		nodir: true,
+	})
+).map((file) => normalizePath(path.join(COMPONENTS_FOLDER, file)));
+
+const stuffToExportRegex = /(@Directive|@Component)\({[\S\s]+?}\)\s*export class ([a-zA-Z]*)/g;
```
Comment (andreascorti) [inline angular/lib/scripts/verify-module.mjs:22]: We have a use case of "empty" `@Directives()` in the headless, but I think your regex is more than enough for components (in general you have at least the `selector`).

**Review by MarkoOleksiyenko — APPROVED** [review]
lgtm, thanks for the PR @quentinderoubaix

**Review by andreascorti — APPROVED** [review]
LGTM, nice way to prevent these issues @quentinderoubaix !

**Pull request merged**